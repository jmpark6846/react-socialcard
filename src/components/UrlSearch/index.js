import React from 'react';
import PropTypes from 'prop-types';

const UrlSearch = ({value, onChange, onSubmit}) => 
  <div className="url">
    <form onSubmit={onSubmit}>
      <div className="input-group">
        <input 
          type="text" 
          placeholder="URL을 입력하세요."
          value={value}
          onChange={onChange}
          className="form-control"
          />
        <div className="input-group-append">
          <button className="btn btn-outline-secondary" type="submit">Submit</button>
        </div>
      </div>  
    </form>
  </div>

UrlSearch.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit:PropTypes.func.isRequired,
  value: PropTypes.string,
}

export default UrlSearch;