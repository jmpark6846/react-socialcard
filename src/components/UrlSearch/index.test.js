import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import UrlSearch from '.';

describe('UrlSearch', ()=>{
	it('renders without crashing', () => {
	  const div = document.createElement('div');
	  ReactDOM.render(<UrlSearch onChange={()=>{}} onSubmit={()=>{}}/>, div);
	  ReactDOM.unmountComponentAtNode(div);
	});

	test('has a valid snapshot', () => {
		const component = renderer.create(<UrlSearch onChange={()=>{}} onSubmit={()=>{}} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});