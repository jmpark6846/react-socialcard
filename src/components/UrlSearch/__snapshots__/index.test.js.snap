// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlSearch has a valid snapshot 1`] = `
<div
  className="url"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="input-group"
    >
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="URL을 입력하세요."
        type="text"
        value={undefined}
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-outline-secondary"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
